python/ray/autoscaler/commands.py

```diff
@@ -458,32 +464,46 @@ def rsync(config_file, source, target, override_cluster_name, down):
     if override_cluster_name is not None:
         config["cluster_name"] = override_cluster_name
     config = _bootstrap_config(config)
-    head_node = _get_head_node(
-        config, config_file, override_cluster_name, create_if_needed=False)
 
     provider = get_node_provider(config["provider"], config["cluster_name"])
```

Comment:
Ah, one optimization we can have here is to have everything check if there's a global variable cached provider, instead of instantiating the provider every time.

Reply:
I will look into it, but I haven't seen any mention of a provider cache before

Reply:
Ah, all I'm saying here is - 

```
# in global scope
_cached_provider = None

def rsync(...):
    if _cached_provider:
        provider = _cached_provider
    else:
        global _cached_provider
        provider = get_node_provider(config["provider"], config["cluster_name"])
        _cached_provider = provider
        
    ...

Reply:
No need to do this in this PR btw